src/utils/misc.cpp

```diff
@@ -766,6 +766,28 @@ QString Utils::Misc::genericCSS() {
 }
 
 /**
+ * A dictonary of search engine name to url for aesthetic ui.
+ * @return {QMap<QString,QString>} dictonary of name to query url.
+ */
+QMap<QString,QString> Utils::Misc::getSearchEnginesMap() {
+    QMap<QString,QString> searchEngines;
+    searchEngines.insert("Google", "https://www.google.com/search?q=");
+    searchEngines.insert("Bing", "https://www.bing.com/search?q=");
+    searchEngines.insert("DuckDuckGo", "https://duckduckgo.com/?t=qownnotes&q=");
+    searchEngines.insert("Yahoo", "https://search.yahoo.com/search?p=");
+    searchEngines.insert("DogPile", "http://www.dogpile.com/search/web?q=");
+    searchEngines.insert("Google Scholar", "https://scholar.google.co.il/scholar?q=");
+    searchEngines.insert("Yandex", "https://www.yandex.com/search/?text=");
+    searchEngines.insert("Ask.com", "https://www.ask.com/web?q=");
+    return searchEngines;
+}
+
+//Returns the default search engine
+QString Utils::Misc::getDefaultSearchEngine() {
+    return "DuckDuckGo";
```

Comment:
It looks like you are using the name of the search engine as setting. I don't think that's a wise choice, because the name could change and then the settings of all users would be invalid. If you want you could use numbers of an `enum` with all search engines as setting-key.

Reply:
I had an idea of saving the search engines as a QVector<QPair<QString, QString>> , do you think this will work ?

Reply:
You could also use a QVector of a struct, like with `QVector<HighlightingRule> _highlightingRulesPre;`, but your approach seems to be ok too...